From: Jakub Wilk <jwilk@jwilk.net>
Date: Thu, 27 Oct 2022 15:39:41 +0200
Subject: [PATCH] find-dbgsym-packages: improve error message for missing
 library

Before:

    $ find-dbgsym-packages /usr/lib/firefox-esr/libxul.so > /dev/null
    W: Cannot parse ldd output: '	libmozsandbox.so => not found'
    W: Cannot parse ldd output: '	liblgpllibs.so => not found'
    W: Cannot parse ldd output: '	libmozsqlite3.so => not found'
    W: Cannot parse ldd output: '	libmozgtk.so => not found'
    W: Cannot parse ldd output: '	libmozwayland.so => not found'

After:

    $ find-dbgsym-packages /usr/lib/firefox-esr/libxul.so > /dev/null
    W: Library not found: libmozsandbox.so
    W: Library not found: liblgpllibs.so
    W: Library not found: libmozsqlite3.so
    W: Library not found: libmozgtk.so
    W: Library not found: libmozwayland.so
---
 find-dbgsym-packages | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/find-dbgsym-packages b/find-dbgsym-packages
index 81eb3cc..07789c8 100755
--- a/find-dbgsym-packages
+++ b/find-dbgsym-packages
@@ -251,6 +251,8 @@ sub get_files_from_elf
             push @libs, $1;
         } elsif ($line =~ /^\t(.+) \(0x[0-9a-f]+\)$/) {
             push @libs, $1;
+        } elsif ($line =~ /^\t(.+) => not found$/) {
+            warn "W: Library not found: $1\n";
         } else {
             warn "W: Cannot parse ldd output: '$line'\n";
         }
-- 
2.37.2

